### PR TITLE
:ambulance: pos_product_sync

### PR DESCRIPTION
**Pull request by fedoranvar**
Improved: compatibility with pos_product_available; Fixed:Synchronizing image of product on every update; After update old product
not removed.

### CODE REVIEW

File: pos_product_sync/models/product_product.py [hunk 1]
```diff
@@ -43,6 +47,10 @@ def send_field_updates(self, product_ids, action=''):
         data = {'message': 'update_product_fields', 'action': action, 'product_ids': product_ids}
         self.env['pos.config'].send_to_all_poses(channel_name, data)
 
+    @api.model
+    def from_js(self, product):
```
Comment (KolushovAlexandr) [inline pos_product_sync/models/product_product.py]: What is it?

File: pos_product_sync/models/product_product.py [hunk 2]
```diff
@@ -54,3 +62,18 @@ def change_product_qty(self):
...
+
+
+class PosOrder(models.Model):
+
+    _inherit = "pos.order"
+
+    # Compatibility with pos_product_available
+    def _process_order(self, pos_order):
+        res = super(PosOrder, self)._process_order(pos_order)
+        order = self.create(self._order_fields(pos_order))
```
Comment (KolushovAlexandr) [inline pos_product_sync/models/product_product.py]: You duplicate the ``create_from_ui method`` functionality and destroy accounting as well. Do not create a new order, you can find all necessary data from the ``pos_order`` dictionary.

File: pos_product_sync/static/src/js/pos_product_sync.js [hunk 3]
```diff
@@ -18,11 +18,15 @@ odoo.define('pos_product_sync.pos', function (require) {
         on_barcode_updates: function(data){
             var self = this;
             if (data.message === 'update_product_fields' && data.product_ids && data.product_ids.length) {
```
Comment (KolushovAlexandr) [inline pos_product_sync/static/src/js/pos_product_sync.js]:
```suggestion
            var product_ids = data && data.product_ids && _.uniq(data.product_ids);
            if (product_ids && data.message === 'update_product_fields' && product_ids.length) {
```

Comment (KolushovAlexandr) [reply pos_product_sync/static/src/js/pos_product_sync.js]: and update ``data.product_ids`` underneath ofcourse.

File: pos_product_sync/static/src/js/pos_product_sync.js [hunk 4]
```diff
@@ -75,6 +79,15 @@ odoo.define('pos_product_sync.pos', function (require) {
...
             this.update_orderlines(product_ids);
+        },
+
+        update_product_related_templates_images: function(product_ids) {
+            var self = this;
+
+            // Product Item
+            var product_screen = this.gui.screen_instances.products;
+            var product_list_widget = product_screen.product_list_widget;
+            product_list_widget.update_product_item(product_ids);
```
Comment (KolushovAlexandr) [inline pos_product_sync/static/src/js/pos_product_sync.js]: dupliated code. dont do that.

File: pos_product_sync/static/src/js/pos_product_sync.js [hunk 5]
```diff
@@ -75,6 +79,15 @@ odoo.define('pos_product_sync.pos', function (require) {
 
             // Orderlines
             this.update_orderlines(product_ids);
+        },
+
+        update_product_related_templates_images: function(product_ids) {
```
Comment (KolushovAlexandr) [inline pos_product_sync/static/src/js/pos_product_sync.js]: Strange, you update here everything except orderlines. Better leave one function as ``update_templates_related_to_product_fields`` and update images with ``update_image`` before  ``update_templates_related_to_product_fields``  is called on the line 25

File: pos_product_sync/models/product_product.py (hunk 1 above)
Comment (fedoranvar) [reply pos_product_sync/models/product_product.py]: Left from playing around at the beginning

File: pos_product_sync/models/product_product.py (hunk 2 above)
Comment (fedoranvar) [reply pos_product_sync/models/product_product.py]: Done

File: pos_product_sync/static/src/js/pos_product_sync.js (hunk 3 above)
Comment (fedoranvar) [reply pos_product_sync/static/src/js/pos_product_sync.js]: I think, it's resolved now

File: pos_product_sync/static/src/js/pos_product_sync.js (hunk 5 above)
Comment (fedoranvar) [reply pos_product_sync/static/src/js/pos_product_sync.js]: done

File: pos_product_sync/doc/index.rst [hunk 6]
```diff
@@ -11,5 +11,12 @@ Installation
...
-* Set **Product Synchronized Fields**
+* Go to ``[Inventory] >> Configuration >> Settings``
+* Check ``Attribute and Variants``
+* In ``Point of Sale Product Synchronization Fields`` add fields that needed to be synchronized
+
+Usage
+=====
+* Go to ``[Point of Sale] >> Catalog >> Product Variants``
+* Choose and edit the product you need
+
```
Comment (KolushovAlexandr) [inline pos_product_sync/doc/index.rst]: add here the result line

**Pull request merged**